### DISCUSSION TITLE
Cilk and RISC-V

### DISCUSSION THREAD

**Discussion by tdysart** [OpenCilk/opencilk-project#226] [Q&A] [answered]
Before trying this out ourselves, has anyone tried to use OpenCilk on the RISC-V architecture and/or tried to build OpenCilk from scratch using a RISC-V toolchain?

This is the base RISC-V version of gcc that we would be using: https://github.com/riscv-collab/riscv-gnu-toolchain

**Comment by VoxSciurorum** [comment, marked as answer]
I don't expect OpenCilk to work on RISC-V.  The compiler should build but it will not be able to generate parallel code.

Internally, `cilk_spawn` uses the `__builtin_setjmp` function which is only available on ARM, x86, and PowerPC.  Because this is a compiler builtin it doesn't matter if gcc supports it.  The version of llvm used by OpenCilk has to support it.   I have a plan to remove this dependency or at least make porting to a new architecture easier.  There is no timeline for this work.

**Reply by tdysart** [reply]
Thanks for the response.
